src/client.ts

```diff
@@ -73,7 +73,14 @@ export class RestClientRequest<ReturnType = any> {
         return this.withHeader('Content-Type', contentType);
     }
 
-    public async call(): Promise<ReturnType | null> {
+    // Overload definitions
+    public async call(): Promise<ReturnType | null>;
+    public async call(options: { returnRawResponse: false }): Promise<ReturnType | null>;
+    public async call(options: { returnRawResponse: true }): Promise<Response>;
+
+    public async call(options?: { returnRawResponse: boolean }): Promise<ReturnType | null | Response> {
```

Comment:
I would suggest it always returns the Response . We can then add another method if we want an option to just get the return type ? 

Or we can add another method for getting the response ?

Reply:
Good point, I've changed it so it now has a .call and a .parse